homeassistant/components/bunq/sensor.py

```diff
@@ -0,0 +1,153 @@
+from bunq.sdk.exception import ApiException, BunqException
+from bunq.sdk.model.generated import endpoint
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import CONF_API_KEY
+from homeassistant.exceptions import PlatformNotReady
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.event import async_track_time_interval
+
+ICON = "mdi:cash-multiple"
+UPDATE_INTERVAL = 60
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({vol.Required(CONF_API_KEY): cv.string})
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set up bunq sensors."""
+    sensors = []
+
+    try:
+        # setup api
+        api_context = ApiContext(
+            ApiEnvironmentType.PRODUCTION, config.get(CONF_API_KEY), "Home Assistant",
+        )
+        api_context.ensure_session_active()
+        BunqContext.load_api_context(api_context)
+
+        # create sensors
+        for account in get_account_data():
+            sensors.append(BunqBalanceSensor(account))
+    except (ApiException, BunqException) as err:
+        _LOGGER.error(err)
+        raise PlatformNotReady
+
+    async_add_entities(sensors, True)
+
+    # schedule updates for sensors
+    data = BunqData(hass, api_context, sensors)
+    await data.schedule_update(UPDATE_INTERVAL)
+
+
+def get_account_data():
+    """Get active bunq accounts."""
+    active_accounts = []
+
+    accounts = endpoint.MonetaryAccountBank.list().value
+    for account in accounts:
+        if account.status == "ACTIVE":
+            active_accounts.append(account)
+
+    return active_accounts
+
+
+class BunqBalanceSensor(Entity):
+    """Setup bunq balance sensor."""
+
+    def __init__(self, account):
+        """Initialize the sensor."""
+        self.id = account.id_
+        self._name = account.description
+        self._state = float(account.balance.value)
+        self._unit_of_measurement = account.currency
+
+    @property
+    def name(self):
+        """Return the name."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state."""
+        return self._state
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        return self._unit_of_measurement
+
+    @property
+    def icon(self):
+        """Return the entity icon."""
+        return ICON
+
+    @property
+    def should_poll(self):
+        """No polling needed."""
+        return False
+
+    def load_data(self, data):
+        """Update sensor data."""
+        try:
+            old_balance = self._state
+            self._state = data.get(self.id)
+            return self._state == old_balance
+        except KeyError:
+            self._state = None
+            _LOGGER.warning("Count not find key in API results.")
```

Comment:
Do not end your logs with a period. See [style guidelines](https://developers.home-assistant.io/docs/en/development_guidelines.html#log-messages) for reference.
```suggestion
            _LOGGER.warning("Count not find key in API results")
```